local-cluster/tests/local_cluster.rs

```diff
@@ -3452,10 +3572,61 @@ fn run_test_load_program_accounts(scan_commitment: CommitmentConfig) {
     t_scan.join().unwrap();
 }
 
+/// Return the next full snapshot archive info after the cluster's last processed slot
+fn wait_for_next_full_snapshot(
+    cluster: &LocalCluster,
+    snapshot_archives_dir: impl AsRef<Path>,
+) -> FullSnapshotArchiveInfo {
+    match wait_for_next_snapshot(
+        cluster,
+        snapshot_archives_dir,
+        NextSnapshotType::FullSnapshot,
+    ) {
+        NextSnapshotResult::FullSnapshot(full_snapshot_archive_info) => full_snapshot_archive_info,
+        _ => unreachable!(),
+    }
+}
+
+/// Return the next incremental snapshot archive info (and associated full snapshot archive info)
+/// after the cluster's last processed slot
+fn wait_for_next_incremental_snapshot(
+    cluster: &LocalCluster,
+    snapshot_archives_dir: impl AsRef<Path>,
+) -> (IncrementalSnapshotArchiveInfo, FullSnapshotArchiveInfo) {
+    match wait_for_next_snapshot(
+        cluster,
+        snapshot_archives_dir,
+        NextSnapshotType::IncrementalAndFullSnapshot,
+    ) {
+        NextSnapshotResult::IncrementalAndFullSnapshot(
+            incremental_snapshot_archive_info,
+            full_snapshot_archive_info,
+        ) => (
+            incremental_snapshot_archive_info,
+            full_snapshot_archive_info,
+        ),
+        _ => unreachable!(),
+    }
+}
+
+#[derive(Debug)]
+enum NextSnapshotType {
+    FullSnapshot,
+    IncrementalAndFullSnapshot,
+}
+
+#[derive(Debug)]
+enum NextSnapshotResult {
+    FullSnapshot(FullSnapshotArchiveInfo),
+    IncrementalAndFullSnapshot(IncrementalSnapshotArchiveInfo, FullSnapshotArchiveInfo),
+}
+
+/// Return the next snapshot archive infos after the cluster's last processed slot
```

Comment:
As we grow the snapshot utilities for local cluster, I think we should move these to a new module.


1) `wait_for_next_incremental_snapshot`, `wait_for_next_full_snapshot` and `wait_for_next_snapshot()` should be in an `impl` block for `LocalCluster`. I think there's enough stuff here where we can actually create an `impl` block for `LocalCluster` in a separate `local_cluster_snapshot_utils.rs`, as shown here: https://hashnode.com/post/managing-big-rust-impls-by-splitting-them-up-ciutsechq0163p253r8kin48n

2. `NextSnapshotType` and `NextSnapshotResult` can be moved into that separate module as well

Reply:
@carllin The `wait_for_()` functions are just in the test code. Do you want me to move this code out of the tests and into the src?

Reply:
Yeah that works for me, all of the logic in the local_cluster/src is pretty much test-specific anyways

Reply:
@carllin Done! Let me know if the layout is what you were thinking.